keps/sig-architecture/20191023-conformance-without-beta.md

```diff
@@ -0,0 +1,167 @@
+
+- [x] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+The KEP describes a process for ensuring that the Kubernetes components and the Kubernetes conformance tests have no dependencies on beta REST APIs or features.
+
+## Motivation
+
+As the Kubernetes project matures, and is used as the foundation for other projects,
+production distributions are expected to increase in stability, reliability, and consistency.
+The Kubernetes conformance project is an important aspect of ensuring consistency across distributions.
+A production distribution should not be required to enable non-GA features or REST APIs in order to pass conformance tests.
+
+Beta dependencies can be grouped into several categories:
+* Kubernetes components requiring beta REST APIs with no stable alternatives (for example, kubelets use the beta CertificateSigningRequest endpoints for cert rotation)
+* Kubernetes components requiring beta REST APIs with stable alternatives (for example, kube-scheduler accidentally switched to using the beta Events REST API in 1.16)
+* Kubernetes components requiring behavior enabled by beta feature gates
+* Kubernetes conformance tests requiring calls to beta REST APIs
+* Kubernetes conformance tests exercising behavior enabled by beta feature gates
+
+For each category, we should identify and resolve existing dependencies, and prevent future occurrences.
+
+### Goals
+
+* Identify existing beta REST APIs and features required by Kubernetes components so they can be graduated or the dependencies removed
+* Identify existing beta REST APIs and features required by conformance tests so the tests can be rewritten to remove those dependencies
+* Prevent new beta REST APIs and features from being required by Kubernetes components or conformance tests
+* Demonstrate passing conformance tests on a cluster with all beta REST APIs and features disabled
+
+### Non-Goals
```

Comment:
I think it's worth stating that non-GA dependencies (things that aren't kubernetes APIs or k8s feature gate guarded) are out of scope for this effort.

Glancing over https://github.com/kubernetes/kubernetes/blob/master/go.mod you can see a ton of core dependencies with < 1.0 version numbers.

(We might want to tackle these someday, but I think that effort should be decoupled from this)

Reply:
done